platform/fabric/core/generic/committer/event.go

```diff
@@ -16,3 +22,56 @@ type TxEvent struct {
 	CommitPeer     string
 	Err            error
 }
+
+type ChaincodeEvent struct {
+	BlockNumber   uint64
+	TransactionID string
+	ChaincodeID   string
+	EventName     string
+	Payload       []byte
+}
+
+func (chaincodeEvent *ChaincodeEvent) Message() interface{} {
+	return chaincodeEvent
+}
+
+func (chaincodeEvent *ChaincodeEvent) Topic() string {
+	return chaincodeEvent.ChaincodeID
+}
+
+func validChaincodeEvent(event *peer.ChaincodeEvent) bool {
+	return event != nil && len(event.GetChaincodeId()) > 0 && len(event.GetEventName()) > 0 && len(event.GetTxId()) > 0
+}
+
+func getChaincodeEvent(env *common.Envelope, blockNumber uint64) (*ChaincodeEvent, error) {
+	var chaincodeEvent *ChaincodeEvent
+	chaincodeAction, err := protoutil.GetActionFromEnvelopeMsg(env)
+	if err != nil {
+		logger.Errorf("Error getting chaincode actions from envelop: %s", err)
+		return nil, err
```

Comment:
shall we return an error with additional message and stack? We can use errors.Wrap